### PR TITLE
TEST-#1961: speed up TestDataFrameReduction_* test

### LINKED ISSUE #1961: Speed up TestDataFrameReduction_* test

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1961 
- [x] tests added and passing

### CODE REVIEW

File: modin/pandas/test/test_dataframe.py [hunk 1]
```diff
@@ -2968,190 +2968,66 @@ def test_hasattr_sparse(self, data):
...
-        else:
-            modin_result = modin_df.any(axis=None, skipna=skipna, bool_only=bool_only)
-            df_equals(modin_result, pandas_result)
+    def test_all_any(self, skipna, bool_only):
+        data = {
+            "int_col": [1, 2, np.NaN, 4],
+            "float_col": [np.NaN, 9.4, 10.1, np.NaN],
+            "str_col": ["a", np.NaN, "c", "d"],
+            "bool_col": [False, True, np.NaN, np.NaN],
+        }
```
Comment (dchigarev) [inline modin/pandas/test/test_dataframe.py:2994]: Many errors may be hidden when we use small dimensioned data. In that case, the whole frame will be located in a single partition, so we're not testing parallel implementation at all. 

Can we use here more dimensional data? I think something about `256x64` will be enough to locate frame in a few partitions

File: modin/pandas/test/test_dataframe.py [hunk 2]
```diff
@@ -2968,190 +2968,66 @@ def test_hasattr_sparse(self, data):
...
+        modin_df, pandas_df = pd.DataFrame(data), pandas.DataFrame(data)
 
-        try:
-            pandas_result = pandas_df.T.any(
-                axis=axis, skipna=skipna, bool_only=bool_only
+        for method in ["all", "any"]:
```
Comment (dchigarev) [inline modin/pandas/test/test_dataframe.py:2997]: can we move it to `pytest.mark.parametrize`?

File: modin/pandas/test/test_dataframe.py [hunk 3]
```diff
@@ -2968,190 +2968,66 @@ def test_hasattr_sparse(self, data):
...
+            new_idx = generate_multiindex(len(modin_df.index))
+            modin_df.index = new_idx
+            pandas_df.index = new_idx
+        else:
+            new_col = generate_multiindex(len(modin_df.columns))
+            modin_df.columns = new_col
+            pandas_df.columns = new_col
 
-                    df_equals(modin_multi_level_result, pandas_multi_level_result)
+        for method in ["all", "any"]:
```
Comment (dchigarev) [inline modin/pandas/test/test_dataframe.py:3025]: same about moving it to a parameters of test function

File: modin/pandas/test/test_dataframe.py [hunk 4]
```diff
@@ -2968,190 +2968,66 @@ def test_hasattr_sparse(self, data):
...
-            df_equals(modin_result, pandas_result)
-
-        try:
-            pandas_result = pandas_df.T.any(
-                axis=None, skipna=skipna, bool_only=bool_only
+            # transposed
+            eval_general(
+                modin_df,
+                pandas_df,
+                lambda df: getattr(df.T, method)(skipna=skipna, bool_only=bool_only),
```
Comment (dchigarev) [inline modin/pandas/test/test_dataframe.py:3007]: I think we also can move transpose-or-not-transpose variable to a function parameter, to explicitly see in a pytest error message which case is failed

File: modin/pandas/test/test_dataframe.py [hunk 5]
```diff
@@ -2968,190 +2968,66 @@ def test_hasattr_sparse(self, data):
...
-            modin_result = modin_df.all(axis=None, skipna=skipna, bool_only=bool_only)
-            df_equals(modin_result, pandas_result)
+    @pytest.mark.parametrize("axis", [None, 0, 1])
+    def test_all_any_default(self, data, axis):
+        modin_df, pandas_df = pd.DataFrame(data), pandas.DataFrame(data)
 
-        try:
-            pandas_result = pandas_df.T.all(
-                axis=axis, skipna=skipna, bool_only=bool_only
+        for method in ["all", "any"]:
```
Comment (dchigarev) [inline modin/pandas/test/test_dataframe.py:2975]: same about moving method name to a test parameter

**Review by dchigarev — CHANGES_REQUESTED** [review]
@anmyachev left some comments

File: modin/pandas/test/test_dataframe.py (hunk 1 above)
Comment (anmyachev) [reply modin/pandas/test/test_dataframe.py:2994]: Is there a way to force set the number of partitions?

Comment (dchigarev) [reply modin/pandas/test/test_dataframe.py:2994]: it's already explicitly setted at the beginning of the tests
https://github.com/modin-project/modin/blob/843816ecbccb6435ba7f29688301f4198277cd27/modin/pandas/test/test_dataframe.py#L67

Comment (anmyachev) [reply modin/pandas/test/test_dataframe.py:2994]: I'm talking about being able to get multiple partitions on a small dataset.

Comment (dchigarev) [reply modin/pandas/test/test_dataframe.py:2995]: No.
For default minimum chunk size is `32x32` (and that default value partition manager uses to [compute chunk size based on `DEFAULT_NPARTITIONS`](https://github.com/modin-project/modin/blob/843816ecbccb6435ba7f29688301f4198277cd27/modin/engines/base/frame/partition_manager.py#L373-L399)) and we have no ability to change it
https://github.com/modin-project/modin/blob/843816ecbccb6435ba7f29688301f4198277cd27/modin/data_management/utils.py#L33-L41

File: modin/pandas/test/test_dataframe.py [hunk 6]
```diff
@@ -3003,316 +3001,136 @@ def test_hasattr_sparse(self, data):
...
-                modin_df.all(axis=None, skipna=skipna, bool_only=bool_only)
-        else:
-            modin_result = modin_df.all(axis=None, skipna=skipna, bool_only=bool_only)
-            df_equals(modin_result, pandas_result)
+    def test_all_any_specific(self, bool_only, method):
+        data = {
+            "float_col": [np.NaN, 9.4, 10.1, np.NaN],
+            "str_col": ["a", np.NaN, "c", "d"],
+            "bool_col": [False, True, True, False],
+        }
```
Comment (dchigarev) [inline modin/pandas/test/test_dataframe.py:3029]: it seems that it is a commonly used data set (repeats 5 times in your RP), maybe we should move it to the `test/utils.py`?

**Comment by dchigarev** [comment]
@anmyachev LGTM.
Let's wait till the tests passes

File: modin/pandas/test/test_dataframe.py (hunk 6 above)
Comment (anmyachev) [reply modin/pandas/test/test_dataframe.py:3029]: will do

File: modin/pandas/test/test_dataframe.py [hunk 7]
```diff
@@ -3328,279 +3136,144 @@ def test_describe(self, data):
...
-        else:
-            modin_result = modin_df.T.min(
-                axis=axis, skipna=skipna, numeric_only=numeric_only
-            )
-            df_equals(modin_result, pandas_result)
-
-    @pytest.mark.skipif(
-        os.name == "nt",
-        reason="Windows has a memory issue for large numbers on this test",
-    )
```
Comment (dchigarev) [inline modin/pandas/test/test_dataframe.py:3523]: Is this issue was fixed?

Comment (anmyachev) [reply modin/pandas/test/test_dataframe.py:3523]: I am not sure, but this issue in current state is not reproducible.

Comment (anmyachev) [reply modin/pandas/test/test_dataframe.py:3523]: Returned this mark, since my goal is not to fix the tests.

**Review by dchigarev — APPROVED** [review]

**Pull request merged**